### DISCUSSION TITLE
Are fields on the KeyDirective fieldset nullable?

### DISCUSSION THREAD

**Discussion by rudrasrivastava** [ExpediaGroup/graphql-kotlin#1261] [Q&A]
Hi, I have a combination of nullable and non-nullable fields on the KeyDirective fieldset. I could not find any documentation regarding if certain fields on the key directive can be nullable given there is atleast one valid non-nullable key. I tried it out and got a "field not in response" error for those fields that are nullable. Would appreciate any clarity on the same.

**Comment by smyrick** [comment]
The federation spec doesn't mention anything specifically about nullability: https://www.apollographql.com/docs/federation/federation-spec/#key

So you could I suppose, but as you mentioned yourself unique identifiers should really be unique and required for a data set perspective
